src/compiler/checker.ts

```diff
@@ -4379,6 +4379,8 @@ module ts {
                     }
                 }
 
+                Debug.assert(!!downfallType, "If there is no common supertype, each type should have a downfallType");

```

Comment:
It *is* supposed to be downfallType. It *is* supposed to sound like it describes revolutions and lost wars, but in particular tragic heroes.

The downfallType is the type that caused a particular candidate to not be the common supertype. So if it weren't for this one downfallType, the type in question could have been the common supertype.

Note that downfallType is the first type that is not a subtype of type. It's not necessarily the only one.